TLM/TLM/Manager/Impl/LaneConnectionManagerData/ConnectionDataBase.cs

```diff
@@ -0,0 +1,186 @@
+namespace TrafficManager.Manager.Impl.LaneConnectionManagerData {
+    using CSUtil.Commons;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using TrafficManager.Util;
+    using TrafficManager.Util.Extensions;
+
+    internal class ConnectionDataBase : Dictionary<LaneEnd, LaneConnectionData[]> {
```

Comment:
Add constructor to use correct Comparer since the whole thing is a Dictionary:
```cs
public ConnectionDataBase(): base(LaneEnd.LaneIdStartNodeComparer) { }
```